### PR TITLE
Fixed Feat: Implement Shimmer while user details are being loaded

### LINKED ISSUE #2412: Feat: Implement Shimmer while user details are being loaded
**Summary:**

I would love to implement a shimmer that is displayed while the details are being loaded. 

**Steps to reproduce:**

When you open the mobile application then the layout showing the user image and name are empty at first because they are being loaded. 
At such a point a shimmer effect would be good to have.

### PR DESCRIPTION
**Pull request by kmanikanta335**
Fixes #2412 

 Screenshots of UI changes.

https://github.com/openMF/mifos-mobile/assets/118070186/747c140f-e653-451d-ad05-4a485c8f5d9c 

https://github.com/openMF/mifos-mobile/assets/118070186/36a2e396-25ed-4254-b548-64bd584c8b9d

Please make sure these boxes are checked before submitting your pull request - thanks!

- ✅  Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- ✅  Run the unit tests with `./gradlew check` to make sure you didn't break anything

- ✅  If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by kmanikanta335** [comment]
@PratyushSingh07 pls review it

File: app/build.gradle [hunk 1]
```diff
@@ -203,5 +203,8 @@ dependencies {
     debugImplementation "androidx.compose.ui:ui-tooling:$rootProject.composeVersion"
     implementation "androidx.compose.material3:material3:$rootProject.materialVersion"
     implementation "androidx.lifecycle:lifecycle-viewmodel-compose:$rootProject.lifecycleVersion"
+
+    implementation 'com.facebook.shimmer:shimmer:0.5.0'
```
Comment (PratyushSingh07) [inline app/build.gradle]: add the version to ext block and add a comment that will describe the dependency

File: app/src/main/java/org/mifos/mobile/ui/fragments/HomeOldFragment.kt [hunk 2]
```diff
@@ -271,7 +275,9 @@ class HomeOldFragment : BaseFragment(), OnRefreshListener {
                         showError(getString(it.errorMessage))
                     }
                     is HomeUiState.UserDetails -> {
-                        hideProgress()
+                        binding.shimmerViewContainer.stopShimmer()
+                        binding.detailsOfCustomer.visibility = View.VISIBLE
+                        binding.shimmerViewContainer.visibility = View.INVISIBLE
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/fragments/HomeOldFragment.kt]: these 3 can be put into a function

**Review by PratyushSingh07 — CHANGES_REQUESTED** [review]
Also make sure to follow the commit guidelines

File: app/build.gradle [hunk 3]
```diff
@@ -203,5 +203,9 @@ dependencies {
     debugImplementation "androidx.compose.ui:ui-tooling:$rootProject.composeVersion"
     implementation "androidx.compose.material3:material3:$rootProject.materialVersion"
     implementation "androidx.lifecycle:lifecycle-viewmodel-compose:$rootProject.lifecycleVersion"
+
+    // Shimmer effect dependencies
```
Comment (PratyushSingh07) [inline app/build.gradle]: `Shimmer Effect` should be enough

File: app/src/main/java/org/mifos/mobile/ui/fragments/HomeOldFragment.kt [hunk 4]
```diff
@@ -361,6 +363,15 @@ class HomeOldFragment : BaseFragment(), OnRefreshListener {
         startActivity(Intent(activity, UserProfileActivity::class.java))
     }
 
+    /**
+     * Stops the Shimmer effect and shows user details
+     */
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/fragments/HomeOldFragment.kt]: we dont need this , the name of the function is self explanatory

File: app/src/main/java/org/mifos/mobile/ui/fragments/RecentTransactionsFragment.kt [hunk 5]
```diff
@@ -70,11 +70,13 @@ class RecentTransactionsFragment : BaseFragment(), OnRefreshListener {
                 is RecentTransactionUiState.Loading -> showProgress()
                 is RecentTransactionUiState.RecentTransactions -> {
                     hideProgress()
+                    Toast.makeText(context,"hello",Toast.LENGTH_SHORT).show()
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/fragments/RecentTransactionsFragment.kt]: dont push changes of other issues in this one. Open another PR that will solve #2442

**Review by PratyushSingh07 — CHANGES_REQUESTED** [review]

**Review by PratyushSingh07 — APPROVED** [review]
LGTM

**Comment by PratyushSingh07** [comment]
make sure to change the commit message @kmanikanta335

**Comment by kmanikanta335** [comment]
@PratyushSingh07 what should I change it to

**Comment by PratyushSingh07** [comment]
> @PratyushSingh07 what should I change it to

just follow the guidelines

**Comment by kmanikanta335** [comment]
@PratyushSingh07 when will it gets merged

**Comment by PratyushSingh07** [comment]
> @PratyushSingh07 when will it gets merged

You still haven't followed the commit guidelines. It will be merged only after the admins approval

**Comment by kmanikanta335** [comment]
@PratyushSingh07 i hope now its fine

**Comment by therajanmaurya** [comment]
Please resolve the conflicts.

**Comment by kmanikanta335** [comment]
@therajanmaurya i hope now it fine

File: app/build.gradle (hunk 1 above)
Comment (PratyushSingh07) [reply app/build.gradle]: you still havent implemented this change @kmanikanta335

**Review by PratyushSingh07 — CHANGES_REQUESTED** [review]
Format your code with `ctrl+shift+L` if  you haven't already. Also I will be converting this PR to draft . You can make it ready for review after you have implemented the changes that i requested

**Comment by PratyushSingh07** [comment]
@kmanikanta335 squash your commits and make sure that your final commit follow contribution guidelines . I am again converting it to draft

**Pull request closed without merging**